Assembler.Core/Entities/BaseMessageInAssembly.cs

```diff
@@ -0,0 +1,16 @@
+﻿using Assembler.Core.Enums;
+
+namespace Assembler.Core.Entities
+{
+    public abstract class BaseMessageInAssembly : BaseMessage
+    {
+        public ReleaseReason ReleaseReason;
+        public bool MiddleReceived;
+
+        protected BaseMessageInAssembly()
```

Comment:
Note that you're forcing the developer to use the `BaseMessage` c'tor that has no parameters (is this intentional?)

Reply:
Yes, I removed the one where you can set your own guid, I don't think the user should have that kind of access

Reply:
👍